File: src/sorting_algo/BubbleSort.java
package sorting_algo;

public class BubbleSort {

    public static void main(String[] args){
        int arr[] = { 64, 34, 25, 12, 22, 11, 90 };
        sorting(arr);
        System.out.print("Sorted array: ");
        SortedArrayPrint(arr);
    }

    static void sorting(int[] arr){
        boolean swaap;
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            swaap = false;
            for (int j=0; j < n-i-1; j++){
                if(arr[j]> arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    swaap = true;
                }
            }
            if(swaap == false)
                break;
        }
    }

    static void SortedArrayPrint(int[] arr){
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

}

File: BubbleSort.js
const BubbleSort = () => {
  let arr = [64, 34, 25, 12, 22, 11, 90];

  let array = sorting(arr);

  console.log("Sorted array:");
  console.log(array);
};

const sorting = (arr) => {
  let swap;
  let stateAtRow = [];
  for (let i = 0; i < arr.length; i++) {
    swap = false;
    for (let j = 0; j < arr.length - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        let temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        swap = true;
      }
    }
    if (i + 1 === arr.length) {
      stateAtRow = [...arr];
    }

    if ((swap = false)) break;
  }

  return stateAtRow;
};

BubbleSort();
